### PR TITLE
set max_steps for lr decay through config

### PR DESCRIPTION
**Pull request by anmolgupt**
# What does this PR do ?

Allows setting lr decay steps through config that could be different from the trainer.max_steps

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by titu1994 — COMMENTED** [review]
We normally will not accept such a experimental code inside core. This raises the possibility of users mistake being ignored with a logging message. 

Please discuss with @ericharper / @okuchaiev and if they agree to accept the PR, it can be merged.

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py [hunk 1]
```diff
@@ -390,7 +401,8 @@ def configure_optimizers(self):
             assert self._trainer.max_steps is not None, "'max_steps' is missing in trainer config."
             if hasattr(self._cfg.optim, 'sched'):
                 sched_config = self._cfg.optim.sched
-                sched_config['max_steps'] = self._trainer.max_steps
+                if 'max_steps' not in sched_config:
+                    sched_config['max_steps'] = self._trainer.max_steps
```
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_base_model.py:405]: Looks fine but can you change the name of `max_steps`? Something to help describe the difference between your `max_steps` and `trainer.max_steps`

Comment (anmolgupt) [reply nemo/collections/nlp/models/language_modeling/megatron_base_model.py:405]: maybe decay_steps?

File: nemo/collections/nlp/parts/megatron_lr_schedulers.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from nemo.core.optim.lr_scheduler import AVAILABLE_SCHEDULERS, CosineAnnealing
```
Comment (ericharper) [inline nemo/collections/nlp/parts/megatron_lr_schedulers.py:1]: Please add the copyright here, you can get it from any file in nemo and set the year to 2023

**Review by titu1994 — APPROVED** [review]
Looks much better, thanks !

**Pull request merged**